tests/e2e_common_endpoints.py

```diff
@@ -209,23 +210,12 @@ def run_large_message_test(
                         get_main_interface_errors()[metrics_name] == before_errors_count
                     )
 
-    # TLS libraries usually have 16K internal buffers, so we start at
-    # 1K and move up to 1M and make sure they can cope with it.
-    # Starting below 16K also helps identify problems (by seeing some
-    # pass but not others, and finding where does it fail).
-    msg_sizes = [2**n for n in range(10, 20)]
-    msg_sizes.extend(
-        [
-            args.max_http_body_size // 2,
-            args.max_http_body_size - 1,
-            args.max_http_body_size,
-            args.max_http_body_size + 1,
-            args.max_http_body_size * 2,
-            args.max_http_body_size * 200,
-        ]
-    )
+    def get_sizes(n):
+        ns = [n // 2, n - 10, n - 1, n, n + 1, n + 10, n * 2, n * 20]
+        random.shuffle(ns)
+        return ns
```

Comment:
One more fix to try and rule out this kind of order-dependent issue - use the same (relative) sizes for each cap, and test them in a random order.